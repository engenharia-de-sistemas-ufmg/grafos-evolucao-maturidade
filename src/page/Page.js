import React from 'react';
import PropTypes from 'prop-types';
import JsonEditor from '../components/JsonEditor/index';
import GraphViewer from '../components/GraphViewer/index';
import InitialPage from '../components/InitialPage/index';
import './Page.css';

const Page = ({ loaded }) => {
  if (!loaded) {
    return (<InitialPage />);
  }

  return (
    <div>
      <JsonEditor />
      <GraphViewer />
    </div>
  );
};

Page.propTypes = {
  loaded: PropTypes.bool.isRequired,
};

export default Page;
