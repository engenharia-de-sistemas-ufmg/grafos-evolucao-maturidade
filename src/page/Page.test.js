import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import Page from './Page';

describe('Page', () => {
  Enzyme.configure({ adapter: new Adapter() });

  test('should match snapshot - loaded', () => {
    const loaded = true;
    const component = shallow(
      <Page loaded={loaded} />,
    );
    expect(toJson(component)).toMatchSnapshot();
  });

  test('should match snapshot - unloaded', () => {
    const loaded = false;
    const component = shallow(
      <Page loaded={loaded} />,
    );
    expect(toJson(component)).toMatchSnapshot();
  });
});
