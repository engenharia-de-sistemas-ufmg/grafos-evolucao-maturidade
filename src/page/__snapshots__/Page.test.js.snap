// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page should match snapshot - loaded 1`] = `
<div>
  <Connect(JsonEditor) />
  <Connect(GraphViewer) />
</div>
`;

exports[`Page should match snapshot - unloaded 1`] = `<Connect(InitialPage) />`;
